h/static/scripts/cross-frame-discovery.coffee

```diff
@@ -0,0 +1,108 @@
+class CrossFrameDiscovery
```

Comment:
Again, I think I prefer the terser `Discovery`, or possibly `FrameDiscovery` at a push.